### PR TITLE
Add missing fields to interaction and webhook

### PR DESCRIPTION
**Pull request by Soheab**
## Summary

This PR adds some missing fields to interaction and webhook methods. The following is added:

**Interaction:**

- **.delete_original_message:**
  - `delay`

- **.edit_original_message:**
  - `attachments`
  - `allowed_mentions`
  - `delete_after`
 
**InteractionReponse:**

- **send_message:**
  - `allowed_mentions`
  - `delete_after`
  - `file`
  - `files`

- **.edit_message:**
  - `allowed_mentions`
  - `attachments`
  - `delete_after`
  - `file`
  - `files`

**InteractionMessage.edit:**
- `allowed_mentions`
- `attachments`
- `delete_after`
 
**WebhookMessage.edit:**
- `attachments`
- `delete_after`

**Webhook:**
- **.send:**
  - documented `delete_after`
- **.edit_message:**
  - `attachments`
  - `delete_after`
- **.delete_message:**
  - `delay`

**SyncWebhookMessage:**
- **edit:**
  - `attachments`
  -  `delete_after`

**SyncWebhook:**
- **edit_message:**
  - `attachments`
  - `delete_after`
- **delete_message:**
  - `delay`

Also added support for file descriptions to send from webhooks/interactions.

## Checklist

- [] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/webhook/async_.py [hunk 1]
```diff
@@ -485,35 +494,45 @@ def handle_message_parameters(
...
-        if len(files) == 1:
-            file = files[0]
+        attachements = payload.get("attachments", [])
+
+        for index, file in enumerate(files):
+            to_append: Dict[str, Union[int, str]] = {"id": index, "filename": file.filename}  # type: ignore | filename can't be None.
+            if file.description is not None:
+                to_append["description"] = file.description
+
+            attachements.append(to_append)  # type: ignore | attachments isn't None.
```
Comment (ow0x) [inline discord/webhook/async_.py:515]: spelling typo: `attachements` to `attachments`?

File: discord/webhook/async_.py [hunk 2]
```diff
@@ -485,35 +494,45 @@ def handle_message_parameters(
...
-                    {
-                        "name": f"file{index}",
-                        "value": file.fp,
-                        "filename": file.filename,
-                        "content_type": "application/octet-stream",
-                    }
-                )
 
-    return ExecuteWebhookParameters(payload=payload, multipart=multipart, files=files)
+        payload["attachments"] = attachements
```
Comment (ow0x) [inline discord/webhook/async_.py:525]: spelling typo: `attachements` to `attachments`?

Comment (Soheab) [reply discord/webhook/async_.py]: thank you! all of them are resolved.

File: docs/whats_new.rst [hunk 3]
```diff
@@ -102,7 +113,7 @@ New Features
             print(f"Permissions for {command}: ", ", ".join(required_permissions.keys())) 
         else:
             print(f"{command} does not use the commands.has_permissions decorator.")
-
+
```
Comment (Chiggy-Playz) [inline docs/whats_new.rst:116]: Why random space here?

**Comment by GnomedDev** [comment]
After a ton of review, I think I'll finally merge this

**Pull request merged**